lib/Styled.ts

```diff
@@ -0,0 +1,42 @@
+import styled from 'styled-components';
+import { Elements } from './Attributes';
+
+export interface ComponentTheme {}
+
+export interface DivTheme {
+    colorTextRest: string;
+    colorBackground: string;
+}
+export interface ButtonTheme {
+    colorRest: string;
+    colorHover: string;
+    colorDisabled: string;
+    colorTextRest: string;
+    colorTextDisabled?: string;
+}
+
+const div = styled(Elements.div)`
+    &&&&&&& {
+        color: ${(props: DivTheme) => props.colorTextRest};
+        background-color: ${(props: DivTheme) => props.colorBackground};
+    }
+`;
+
+const button = styled(Elements.button)`
+    &&&&& {
+        color: ${(props: { theme: ButtonTheme}) => props.theme.colorTextRest};
```

Comment:
why some of these have `props: { theme: Interface }` while others are `props: Interface` can we make them consistent?